tests/test_metrics/test_metrics_utils.py

```diff
@@ -21,3 +26,160 @@ def test_compute_metrics(train_test_dfs: Tuple[TSDataset, TSDataset]):
     ]
     result = compute_metrics(metrics=metrics, y_true=true_df, y_pred=forecast_df)
     np.testing.assert_array_equal(sorted(expected_keys), sorted(result.keys()))
+
+
+@pytest.fixture
+def metrics_df_with_folds() -> pd.DataFrame:
+    df = pd.DataFrame(
+        {
+            "segment": ["segment_0"] * 3 + ["segment_1"] * 3 + ["segment_2"] * 3,
+            "MAE": [1.0, 2.0, 3.0, 2.0, 3.0, 4.0, 3.0, 4.0, 5.0],
+            "MSE": [2.0, 3.0, 4.0, 3.0, 4.0, 5.0, 5.0, 6.0, 7.0],
+            "fold_number": [0, 1, 2, 0, 1, 2, 0, 1, 2],
+        }
+    )
+    return df
+
+
+@pytest.fixture
+def metrics_df_no_folds(metrics_df_with_folds) -> pd.DataFrame:
+    df = metrics_df_with_folds
+    df = df.groupby("segment").mean().reset_index().drop("fold_number", axis=1)
+    return df
+
+
+@pytest.fixture
+def aggregated_metrics_df() -> Dict[str, Any]:
+    result = {
+        "MAE_median": 3.0,
+        "MAE_mean": 3.0,
+        "MAE_std": 0.816496580927726,
+        "MAE_size": 3.0,
+        "MAE_percentile_5": 2.1,
+        "MAE_percentile_25": 2.5,
+        "MAE_percentile_75": 3.5,
+        "MAE_percentile_95": 3.9,
+        "MSE_median": 4.0,
+        "MSE_mean": 4.333333333333333,
+        "MSE_std": 1.247219128924647,
+        "MSE_size": 3.0,
+        "MSE_percentile_5": 3.1,
+        "MSE_percentile_25": 3.5,
+        "MSE_percentile_75": 5.0,
+        "MSE_percentile_95": 5.8,
+    }
+    return result
+
+
+@pytest.fixture
+def metrics_df_with_folds_with_missing() -> pd.DataFrame:
+    df = pd.DataFrame(
+        {
+            "segment": ["segment_0"] * 3 + ["segment_1"] * 3 + ["segment_2"] * 3,
+            "MAE": [None, 2.0, 3.0, 2.0, 3.0, 4.0, 3.0, 4.0, 5.0],
+            "MSE": [2.0, 3.0, 4.0, 3.0, 4.0, 5.0, 5.0, 6.0, 7.0],
+            "fold_number": [0, 1, 2, 0, 1, 2, 0, 1, 2],
+        }
+    )
+    return df
+
+
+@pytest.fixture
+def metrics_df_no_folds_with_missing(metrics_df_with_folds_with_missing) -> pd.DataFrame:
+    df = metrics_df_with_folds_with_missing
+    df = (
+        df.groupby("segment")
+        .apply(lambda x: x.mean(skipna=False, numeric_only=False))
+        .reset_index()
+        .drop("fold_number", axis=1)
+    )
+    return df
+
+
+@pytest.fixture
+def aggregated_metrics_df_with_missing() -> Dict[str, Any]:
+    result = {
+        "MAE_mean": 3.5,
+        "MAE_median": 3.5,
+        "MAE_std": 0.5,
+        "MAE_size": 2.0,
+        "MAE_percentile_5": 3.05,
+        "MAE_percentile_25": 3.25,
+        "MAE_percentile_75": 3.75,
+        "MAE_percentile_95": 3.95,
+        "MSE_mean": 4.333333333333333,
+        "MSE_median": 4.0,
+        "MSE_std": 1.247219128924647,
+        "MSE_size": 3.0,
+        "MSE_percentile_5": 3.1,
+        "MSE_percentile_25": 3.5,
+        "MSE_percentile_75": 5.0,
+        "MSE_percentile_95": 5.8,
+    }
+    return result
+
+
+@pytest.fixture
+def metrics_df_with_folds_with_full_missing() -> pd.DataFrame:
```

Comment:
We should also add a case when whole segment folds are `None`

Reply:
Do you mean for all folds to be filled with None?